client/object_delete.go

```diff
@@ -119,9 +119,12 @@ func (x ResObjectDelete) Tombstone() oid.ID {
 // NeoFS status codes are returned as `error`, otherwise, are included
 // in the returned result structure.
 //
-// Immediately panics if parameters are set incorrectly (see PrmObjectDelete docs).
 // Context is required and must not be nil. It is used for network communication.
 //
+// Return errors:
+//   - [ErrMissingContainer]
+//   - [ErrMissingObject]
```

Comment:
And any of `apistatus` ones, and context, and maybe something else. But yeah, that's the way I'd really like it to be. Let's have them this way here and then we'll see how many of them will still be relevant after #412.

Reply:
sure